db/db.go

```diff
@@ -1,105 +1,94 @@
 package db
 
 import (
+	"context"
 	"sync"
-	"time"
 
 	"github.com/mongodb/amboy"
 	"github.com/pkg/errors"
-	mgo "gopkg.in/mgo.v2"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
 type sessionCache struct {
-	s      *mgo.Session
-	mq     amboy.Queue
-	rq     amboy.Queue
+	client *mongo.Client
+	ctx    context.Context
```

Comment:
Typing this out convinced me to use the application context for cur.Next too. I added checking the request context in between calls to Next.

Reply:
I prefer to use request contexts for deadlines (no point of keeping a connection open if the request timeout anyways), and global contexts are typically a code smell (but there are cases where they are useful). Why did the driver team advice against using the request context?

Reply:
I'm with you on the desire to use the request context.

What I understood from them is that every time the context is cancelled while the query is still ongoing the server takes a small performance hit. If we do it a bunch it could have a deleterious effect on the server. So they recommended we make it a once in a while thing (e.g. the application is going down for restart) rather than an all the time thing (e.g. the request was cancelled).